packages/starknet-snap/src/index.ts

```diff
@@ -279,34 +308,54 @@ export const onHomePage: OnHomePageHandler = async () => {
     // default network is testnet
     let network = STARKNET_SEPOLIA_TESTNET_NETWORK;
 
-    if (state.currentNetwork && state.currentNetwork.chainId !== STARKNET_TESTNET_NETWORK.chainId) {
+    if (
+      state.currentNetwork &&
+      state.currentNetwork.chainId !== STARKNET_TESTNET_NETWORK.chainId
+    ) {
       network = state.currentNetwork;
     }
 
     // we only support 1 address at this moment
     const idx = 0;
     const keyDeriver = await getAddressKeyDeriver(snap);
-    const { publicKey } = await getKeysFromAddressIndex(keyDeriver, network.chainId, state, idx);
+    const { publicKey } = await getKeysFromAddressIndex(
+      keyDeriver,
```

Comment:
sometimes we cast it : `keyDeriver as unknown as BIP44AddressKeyDeriver` (for example in : ./estimateAccountDeployFee.ts:35) sometimes we don't. Is there a reason ?

Reply:
i think it is because the keyDeriver pass in from apiparam or get it by urself